lib/streaming.js

```diff
@@ -58,23 +58,29 @@ exports.StreamType = StreamType;
  * @private
  * @constructor
  * @param {StreamType} type - the type of gRPC stream.
- * @param {ApiCallback} callback - the callback for further API call.
  */
-function StreamProxy(type, callback) {
+function StreamProxy(type) {
   DuplexStream.call(this, {
     objectMode: true,
     readable: type !== StreamType.CLIENT_STREAMING,
     writable: type !== StreamType.SERVER_STREAMING
   });
   this.type = type;
-  this._callback = callback;
   this._writeQueue = [];
   this._isEndCalled = false;
   this._isCancelCalled = false;
   var self = this;
+  this._flowing = false;
   this.on('finish', function() {
     self._onFinish();
   });
+  if (type !== StreamType.CLIENT_STREAMING) {
+    this.on('newListener', function(event) {
+      if (event === 'status' || event === 'metadata' || event === 'response') {
```

Comment:
error isn't necessarily to be here, anyways error could happen.
`resume()` is called when the first `data` listener is added, and I believe potentially those events should be added.